src/data/markdown/docs/30 xk6-browser/01 Get started/03 Running xk6-browser.md

```diff
@@ -0,0 +1,237 @@
+
+<Blockquote mod="note" title="">
+
+You can also use `page.$()` instead of `page.locator()`. You can find the differences between `page.locator()` and `page.$` in the [Locator API documentation](/javascript-api/xk6-browser/locator/).
+
+</Blockquote>
+
+<CodeGroup labels={["script.js"]} lineNumbers={[true]}>
+
+```javascript
+import { chromium } from 'k6/x/browser';
+
+export default function () {
+  const browser = chromium.launch({ headless: false });
+  const page = browser.newPage();
+
+  page
+    .goto('https://test.k6.io/my_messages.php', { waitUntil: 'networkidle' })
+    .then(() => {
+      // Enter login credentials
+      page.locator('input[name="login"]').type('admin');
+      page.locator('input[name="password"]').type('123');
+
+      page.screenshot({ path: 'screenshot.png' });
+    })
+    .finally(() => {
+      page.close();
+      browser.close();
+    });
+}
+```
+
+</CodeGroup>
+
+The preceding code creates and returns a Locator object with the selectors for both login and password passed as arguments. 
+
+Within the Locator API, various methods such as `type()` can be used to interact with the elements. The `type()` method types a text to an input field. 
+
+## Asynchronous operations
+
+As explained previously, the k6 API is synchronous. However, since many browser operations happen asynchronously, and in order to follow the Playwright API more closely, we are working on migrating most xk6-browser methods to be asynchronous as well.
+
+At the moment, methods such as `page.goto()`, `page.waitForNavigation()` and `Element.click()` return [JavaScript promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises), and scripts must be written to handle this properly.
+
+To avoid timing errors or other race conditions in your script, if you have actions that load up a different page, you need to make sure that you wait for that action to finish before continuing.
+
+<CodeGroup labels={["script.js"]} lineNumbers={[true]}>
+
+<!-- eslint-skip -->
+
+```javascript
+import { check } from 'k6';
+import { chromium } from 'k6/x/browser';
+
+export default function () {
+  const browser = chromium.launch({ headless: false });
+  const page = browser.newPage();
+
+  page
+    .goto('https://test.k6.io/my_messages.php', { waitUntil: 'networkidle' })
+    .then(() => {
+      // Enter login credentials and login
+      page.locator('input[name="login"]').type('admin');
+      page.locator('input[name="password"]').type('123');
+      
+      // Wait for asynchronous operations to complete
+      return Promise.all([
+        page.waitForNavigation(),
+        page.locator('input[type="submit"]').click(),
+      ]).then(() => {
+        check(page, {
+          'header': page.locator('h2').textContent() == 'Welcome, admin!',
+        });
+      }).finally(() => {
+        page.close();
+        browser.close();
+      });
+    });
```

Comment:
The `finally` better be after the first `then` (`.goto`'s `then`) since `goto` can also throw an exception.

```suggestion
      });
    }).finally(() => {
      page.close();
      browser.close();
    });
```

Reply:
Aha, i was also wondering what was best practice here, thanks for this @inancgumus!